### PR TITLE
CO-117 Query to get roles/role

### PR DESCRIPTION
**Pull request by JeppeTrip**

### CODE REVIEW

File: backend/Application/Roles/Queries/GetAllRoles/GetAllRolesQuery.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+      return pagesLeft;
+    }
+
+    public System.Linq.IQueryable<Role> PreparePage(IQueryable<Role> query)
+    {
+      if (Skip.HasValue)
+      {
+        return query
+            .OrderBy(x => x.Name)
+            .Where(x => String.Compare(x.Name, Needle) > 0)
```
Comment (objmj-itminds) [inline backend/Application/Roles/Queries/GetAllRoles/GetAllRolesQuery.cs:41]: Have you tried this from Swagger? The String Compare method might not be database context compatible.

File: backend/Application/Roles/Queries/GetRole/GetRoleQuery.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+using AutoMapper;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Application.Roles.Queries.GetRole
+{
+  [AuthorizeAttribute(Domain.Enums.Action.GET_ROLES)]
+  public class GetRoleQuery : IRequest<RoleDto>
+  {
+    public string Name { get; set; }
```
Comment (objmj-itminds) [inline backend/Application/Roles/Queries/GetRole/GetRoleQuery.cs:16]: Roles have an ID let's get details on that instead.

File: backend/Web/Controllers/RoleController.cs [hunk 3]
```diff
@@ -19,5 +22,24 @@ public async Task<ActionResult<RoleIdDto>> UpdateRole(UpdateRoleCommand command)
...
+
+    [HttpGet("{name}")]
+    public async Task<ActionResult<RoleDto>> GetRole([FromRoute] string name)
+    {
+      return await Mediator.Send(new GetRoleQuery { Name = name });
+    }
+
+    [HttpGet("AllRoles")]
+    public async Task<ActionResult<PageResult<RoleDto, string>>> GetAllRole(
+      [FromQuery] string? needle, [FromQuery] int size = 1000, [FromQuery] int? skip = 0
```
Comment (objmj-itminds) [inline backend/Web/Controllers/RoleController.cs:34]: default the needle to an empty string.

File: backend/Application/Roles/Queries/GetAllRoles/GetAllRolesQuery.cs (hunk 1 above)
Comment (JeppeTrip) [reply backend/Application/Roles/Queries/GetAllRoles/GetAllRolesQuery.cs:41]: Just tried using the query from swagger, and it seems to me like it works just fine.

**Review by objmj-itminds — APPROVED** [review]

**Pull request merged**